config/redis.yml

```diff
@@ -1,20 +1,17 @@
 default: &DEFAULT
-  :db: 1
-  :namespace: canary
+  :db: <%= ENV.fetch('STORAGE_REDIS_DB_INDEX') %>
 
 sidekiq:
   <<: *DEFAULT
-  :host: <%= ENV.fetch('REDIS_PORT_6379_TCP_ADDR') %>
-  :port: <%= ENV.fetch('REDIS_PORT_6379_TCP_PORT', 6379) %>
-  :namespace: <%= [ 'canary', 'sidekiq' ].join(':') %>
+  :host: <%= ENV.fetch('STORAGE_REDIS_HOST') %>
+  :port: <%= ENV.fetch('STORAGE_REDIS_PORT', 6379) %>
 
 session: &SESSION
   <<: *DEFAULT
-  :host: <%= ENV.fetch('REDIS_PORT_6379_TCP_ADDR_SESSION') %>
-  :port: <%= ENV.fetch('REDIS_PORT_6379_TCP_PORT_SESSION', 6379) %>
-  :namespace: <%= [ 'canary', 'session' ].join(':') %>
+  :host: <%= ENV.fetch('SESSION_REDIS_HOST') %>
+  :port: <%= ENV.fetch('SESSION_REDIS_PORT', 6379) %>
+  :db: <%= ENV.fetch('SESSION_REDIS_DB_INDEX') %>
   :expires_in: <%= 2.days %>
 
 cache:
   <<: *SESSION
-  :namespace: <%= [ 'canary', 'cache' ].join(':') %>
```

Comment:
just to verify, it's ok to use the same db for session & cache, without namespace config?

Reply:
That's a good question.  I'd love to get out of using namespaces entirely.  I suspect that sessions and cache can live together fine in the same database, but I'm not 100% sure.  We can certainly try it out with the canary app.  I don't know if we want to go so far as to assign an extra redis database per application for the purposes of separating session and cache either.  @Omicron7 any thoughts?

Reply:
I'm almost certain it won't cause problems with the canary app. But that's because it doesn't do much with sessions or caching, so it might not be a good indicator for other apps.

But I'd guess the likelihood of a cache key colliding with a session key is pretty minimal. I'd be ok with living with that risk. But I think it's something you (app devs) would need to be comfortable with, as well.

Reply:
I'm betting rails keys both different to prevent collisions. Lets try it without namespaces.